drivers/nuopc/ocn_map_river_nutrients.F90

```diff
@@ -0,0 +1,148 @@
+module ocn_map_river_nutrients
+
+   use ESMF              , only : ESMF_Mesh, ESMF_MeshCreate, ESMF_FILEFORMAT_ESMFMESH
+   use ESMF              , only : ESMF_Field, ESMF_FieldCreate
+   use ESMF              , only : ESMF_FieldBundle, ESMF_FieldBundleCreate
+   use ESMF              , only : ESMF_FieldBundleAdd, ESMF_FieldBundleGet
+   use ESMF              , only : ESMF_SUCCESS, ESMF_LogFoundError
+   use ESMF              , only : ESMF_MESHLOC_ELEMENT, ESMF_TYPEKIND_R8
+   use nuopc_shr_methods , only : chkerr
+   use shr_kind_mod      , only : r8 => shr_kind_r8, CL => shr_kind_cl, CS => shr_kind_cs
+   use shr_log_mod       , only : errMsg => shr_log_errMsg
+   use mod_io_input      , only : read_map_input_data, field_getfldptr
+   use mod_io_output     , only : io_write
+   use mo_read_rivin     , only : rivflx(:,:,:)
+
+   use mod_xc
+
+   implicit none
+   private
+
+   public :: map_river_nutrients
+
+   character(len=*), parameter :: u_FILE_u = &
+        __FILE__
+
+!===============================================================================
+contains
+!===============================================================================
+
+   subroutine map_river_nutrients(mesh_blom, rc)
+
+      ! input/out variables
+      type(ESMF_Mesh)     , intent(in)  :: mesh_blom
+      integer             , intent(out) :: rc
+
+      ! local variables:
+      character(len=CL)      :: mesh_input_file
+      character(len=CL)      :: filename
+      character(len=3)       :: fldlist_input(7)
+      character(len=CL)      :: mapfile
+      type(ESMF_Mesh)        :: mesh_input
+      type(ESMF_Field)       :: field_blom
+      type(ESMF_FieldBundle) :: fldbun_blom
+      integer                :: nf,n,i,j,ko,l
+      integer                :: jjcpl
+      real(r8), pointer      :: dataptr1d(:)
+      integer                :: nlev
+      integer                :: nfld
+      !-------------------------------------------------------------------------------
+
+      rc = ESMF_SUCCESS
+
+      filename = '/cluster/shared/noresm/inputdata/ocn/blom/bndcon/river_nutrients_GNEWS2000c00_r05_20250220.nc'
+      mesh_input_file = '/cluster/shared/noresm/inputdata/share/meshes/r05_nomask_c110308_ESMFmesh.nc'
+      fldlist_input = (/'DIN','DIP','DSi','DIC','Fe ','DOC','DET'/)
```

Comment:
This currently seems to only support the 'old' river input file - for the `river2omip` the `slDOC` seems to be missing.

Reply:
That is correct. @JorgSchwinger just provided a new input dataset that I can use and I'll add that field.